internal/test/wsu/wsu_test.go

```diff
@@ -98,32 +98,45 @@ func testAllVMs(t *testing.T) {
 func setupAndTest(t *testing.T, vmNum int) {
 	// Indicate that we can run the test suite on each node in parallel
 	t.Parallel()
-
+	vm := framework.WinVMs[vmNum]
 	// Run the WSU against the VM
-	wsuOut, err := runWSU(framework.WinVMs[vmNum])
-	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, wsuOut)
-
+	wsuOut, err := runWSU(vm)
+	wsuStringOutput := string(wsuOut)
+	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, wsuStringOutput)
+	// TODO: Think of a better way to refactor this function later to get just output that can be consumed by framework.
+	logFileName := t.Name() + "wsu.log"
```

Comment:
@sebsoto - Let me know, if this addresses your concern of running WSU multiple times.

Reply:
Be aware that the test name contains foward slashes, which will cause sub directories to be created.

Reply:
We should be good with that as long as we've VM name being generated for test. Look at the log file that got generated here:

https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/141/pull-ci-openshift-windows-machine-config-operator-master-e2e-wsu/628/artifacts/e2e-wsu/nodes/ip-10-0-19-99.ec2.internal/logs/TestWSU/VM_specific_tests/